clang/lib/AST/ExprConstant.cpp

```diff
@@ -1571,6 +1572,7 @@ static bool isValidIndeterminateAccess(AccessKinds AK) {
     return false;
 
   case AK_IsWithinLifetime:
+  case AK_Dereference:
```

Comment:
Dereferencing an indeterminate value doesn't seem like a valid access? Do you have a test case for this situation?